tests/test_filters.py

```diff
@@ -86,6 +92,50 @@ def test_longdate_datetime_with_tz(self):
         with self.app.test_request_context('/'):
             assert filters.longdate(testdate) == testdate.strftime('%e %B %Y')
 
+    def test_date_filter(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.date, 'yyyy-MM-dd', usertz=False) == self.date.strftime('%Y-%m-%d')
+
+    def test_date_localized_short_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, format='short') == u'1/1/20'
```

Comment:
Problem with using Jan 1: can't tell if this is MDY or DMY. Under what conditions does Babel switch? Pick a test date > 12th of the month and add tests for MDY vs DMY vs YMD. Guessing this will change between `en_US` and `en_GB`/`en_IN`.